src/org/labkey/cds/CDSController.java

```diff
@@ -1370,4 +1373,29 @@ public void setReferrer(String referrer)
         }
 
     }
+
+    @RequiresPermission(ReadPermission.class)
+    public class GetReportsDataAction extends ReadOnlyApiAction<Object>
```

Comment:
Can we get rid of this action and use reports-BrowseDataTree instead? Reports can further be filtered on the client side. Also, only R report types should be included.

Reply:
Good to know, now using reports-BrowseDataTree.